unit-tests/py/test.py

```diff
@@ -54,74 +54,96 @@ def check_failed():
     n_failed_assertions += 1
     test_failed = True
 
-def abort_test():
-    print("Abort was specified in a failed check. Aborting test")
+def abort():
+    print("A was specified in a failed check. Aborting test")
     exit(1)
 
 # Receive an expression which is an assertion. If false the assertion failed.
-def check(exp, abort = False):
+def check(exp, abort_if_failed = False):
     global n_assertions
     n_assertions += 1
     if not exp:
-        print("Check failed, received", end = " ")
-        print(exp)
+        print("Check failed, received", exp)
         check_failed()
         print_stack()
-        if abort:
-            abort_test()
-
+        if abort_if_failed:
+            abort()
+        return False
+    return True
 
 # Receives the resulted value and the expected value and asserts they are equal
-def check_equal(result, expected, abort = False):
+def check_equal(result, expected, abort_if_failed = False):
     check(type(expected) != list)
     global n_assertions
     n_assertions += 1
-    if not result == expected:
-        print("Result was: " + result + "\nBut we expected: " + expected)
+    if result != expected:
+        print("Result was:" + result + "\nBut we expected: " + expected)
         check_failed()
         print_stack()
-        if abort:
-            abort_test()
+        if abort_if_failed:
+            abort()
+        return False
+    return True
 
 # This function should never be reached
-def check_not_reached( abort = False ):
-    check(False, abort)
+def check_not_reached( abort_if_failed = False ):
+    check(False, abort_if_failed)
+
+# This function should be put in except blocks that should not be reached.
+# It's different from check_not_reached because it expects to be in an except block and prints the stack of the error
+# and not the call-stack for this function
+def check_no_exception( abort_if_failed = False ):
```

Comment:
It's not really a check.
If I'm in a block where I expect no exception:
```
except:
    test.catch_no_exception()
```
Is not as readable as:
```
except:
    test.unexpected_exception()
```
I'd say a similar thing for `check_not_reached` -- it's not a check, either. They both fail the test. I'd say call the other `test.unreachable()`?